File: src/main/java/de/dk/api/codeChallenge/notification/repository/NotificationTypeRepository.java
package de.dk.api.codeChallenge.notification.repository;

import de.dk.api.codeChallenge.notification.model.NotificationTypeEntity;
import de.dk.api.codeChallenge.notification.model.enums.Category;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface NotificationTypeRepository extends JpaRepository<NotificationTypeEntity, UUID> {
    NotificationTypeEntity findByNameIgnoreCase(String name);

    List<NotificationTypeEntity> findAllByCategory(Category category);
}

File: src/main/kotlin/de/dkb/api/codeChallenge/notification/repository/NotificationTypeRepository.kt
package de.dkb.api.codeChallenge.notification.repository


import de.dkb.api.codeChallenge.notification.model.enums.Category
import de.dkb.api.codeChallenge.notification.model.NotificationTypeEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface NotificationTypeRepository : JpaRepository<NotificationTypeEntity, UUID> {
    fun findByNameIgnoreCase(name: String): NotificationTypeEntity?
    fun findAllByCategory(category: Category): List<NotificationTypeEntity>
}
